File: docs/vsto/codesnippet/VisualBasic/CallingCodeFromVBA/ThisDocument.vb
<Microsoft.VisualBasic.ComClassAttribute()> <System.Runtime.InteropServices.ComVisibleAttribute(True)> Public Class ThisDocument

    Private Sub ThisDocument_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup

    End Sub

    Private Sub ThisDocument_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown

    End Sub

    '<Snippet1>
    Public Sub CreateTable(ByVal firstColumnHeader As String, _
        ByVal secondColumnHeader As String)

        Me.Paragraphs(1).Range.InsertParagraphBefore()
        Dim table1 As Word.Table = Me.Tables.Add(Me.Paragraphs(1).Range, 2, 2)

        With table1
            .Style = "Table Professional"
            .Cell(1, 1).Range.Text = firstColumnHeader
            .Cell(1, 2).Range.Text = secondColumnHeader
        End With
    End Sub
    '</Snippet1>

End Class

File: docs/vsto/codesnippet/CSharp/worddocument1/ThisDocument.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Xml.Linq;
using Microsoft.Office.Tools.Word;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Office = Microsoft.Office.Core;
using Word = Microsoft.Office.Interop.Word;

namespace WordDocument1
{
    public partial class ThisDocument
    {
        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
            this.Paragraphs[1].Range.Text = "The blue doll is very blue in the face. Which makes me BLuE too. But not red!";
            Test();
        }

        void Test()
        {
            //<Snippet1>
            Word.Range documentRange = this.Application.ActiveDocument.Content;
            documentRange.Find.ClearFormatting();
            documentRange.Find.Execute(FindText: "blue", ReplaceWith: "red", Replace: Word.WdReplace.wdReplaceAll);            
            //</Snippet1>

            //<Snippet2>
            documentRange.Find.ClearFormatting();
            documentRange.Find.Execute("blue", ref missing, ref missing, ref missing, ref missing, ref missing,
                ref missing, ref missing, ref missing, "red", Word.WdReplace.wdReplaceAll, ref missing, ref missing,
                ref missing, ref missing);
            //</Snippet2>

#if DoCompileError
            //<Snippet3>
            Globals.ThisDocument.CheckSpelling(ignoreUppercase: true);
            //</Snippet3>
#endif

            //<Snippet4>
            object ignoreUppercase = true;
            Globals.ThisDocument.CheckSpelling(
                ref missing, ref ignoreUppercase, ref missing, ref missing, ref missing, ref missing,
                ref missing, ref missing, ref missing, ref missing, ref missing, ref missing);
            //</Snippet4>

            //<Snippet5>
            Globals.ThisDocument.InnerObject.CheckSpelling(IgnoreUppercase: true);
            //</Snippet5>
        }

        private void ThisDocument_Shutdown(object sender, System.EventArgs e)
        {
        }

        #region VSTO Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisDocument_Startup);
            this.Shutdown += new System.EventHandler(ThisDocument_Shutdown);
        }

        #endregion
    }
}
